rocAL_pybind/amd/rocal/plugin/tf.py

```diff
@@ -183,16 +184,18 @@ def __next__(self):
                     self.labels = np.zeros(
                         (self.bs) * (self.loader._num_classes), dtype="int32")
                     self.loader.get_one_hot_encoded_labels(
-                        self.labels, device="cpu")
+                        self.labels.ctypes.data_as(
```

Comment:
I think it is better to pass (void *) for both cpu/gpu use cases. The low level function can interpret it as gpu mem pointer